lib/pdc/resource/relation.rb

```diff
@@ -37,9 +38,19 @@ def uri
     # so find(id, release: 'rel') need to work
     def find(id, vars = {})
       raise PDC::ResourceNotFound if id.blank?
-      where(primary_key => id)
-        .where(vars)
-        .find_one!
+      begin
+        where(primary_key => id)
+          .where(vars)
+          .find_one!
+      rescue Faraday::ConnectionFailed => e
+        raise PDC::ConnectionFailed.new e
+      rescue Curl::Err::HostResolutionError => e
+        raise PDC::HostResolutionError.new e
+      rescue Curl::Err::CurlError => e
```

Comment:
We don't want a CurlError since it is leaking the details of the implmentation outside, think how the users code will have to change if we replace curl with something else. `Curl` and `Faraday` are implementation details of the gem that users shouldn't care about.

How about we return `PDC::Error` if it is a `CurlError`

Reply:
I have changed this part, and also keep the PDC's errors raise as the original way, just wrap the non pdc errors, user can use a PDC:Error to catch it.